<?xml version="1.0"?>
<launch>

    <arg name="robot_id" default="robot" />
    <arg name="prefix" default="$(arg robot_id)_" />

    <arg name="serial_port" default="/dev/ttyUSB0" />
    <arg name="baudrate" default="115200" />

    <arg name="motor_frame_id" default="$(arg prefix)base_link"/>
    <arg name="motor_setpoint_topic" default="motor_controller/set_setpoint"/>
    <arg name="motor_state_topic" default="motor_controller/get_state"/>

    <arg name="imu_frame_id" default="$(arg prefix)imu_link" />
    <arg name="imu_topic" default="imu/data" />

    <arg name="leds_service" default="set_leds_signal"/>
    <arg name="leds_signals_config" default="$(find nano_atom_driver)/config/leds_signals.yaml" />


    <group ns="$(arg robot_id)">

        <node pkg="nano_atom_driver" type="nano_driver_node" name="nano_driver" output="screen">

            <param name="motor_frame_id" value="$(arg motor_frame_id)" />
            <param name="motor_setpoint_topic" value="$(arg motor_setpoint_topic)" />
            <param name="motor_state_topic" value="$(arg motor_state_topic)" />

            <param name="imu_frame_id" value="$(arg imu_frame_id)" />
            <param name="imu_topic" value="$(arg imu_topic)" />

            <param name="leds_service" value="$(arg leds_service)" />
            <rosparam command="load" file="$(arg leds_signals_config)" />
       
        </node>

        
        <node pkg="rosserial_python" type="serial_node.py" name="serial_node" output="screen">
            <param name="port" value="$(arg serial_port)"/>
            <param name="baud" value="$(arg baudrate)"/>
        </node>
        

    </group>

</launch>